seatunnel-engine/seatunnel-engine-client/src/main/java/org/apache/seatunnel/engine/client/job/ClientJobProxy.java

```diff
@@ -117,10 +115,6 @@ public JobResult waitForJobCompleteV2() {
             throw new RuntimeException(e);
         }
         LOGGER.info(String.format("Job (%s) end with state %s", jobId, jobResult.getStatus()));
-        if (StringUtils.isNotEmpty(jobResult.getError())
-                || jobResult.getStatus().equals(JobStatus.FAILED)) {
-            throw new SeaTunnelEngineException(jobResult.getError());
-        }
```

Comment:
If client never throw exception even job failed. The shell status code always return 0, does not meet shell standards. Also client side will not get job result error msg. We need test case to cover it.

Reply:
> If client never throw exception even job failed. The shell status code always return 0, does not meet shell standards. Also client side will not get job result error msg. We need test case to cover it.

Done, I have added test cases.